playbooks/roles/compatibility-layer/tasks/add_overlay.yml

```diff
@@ -0,0 +1,46 @@
+# Add a custom overlay to the Gentoo Prefix installation.
+---
+- name: Install equery command (dependency for the portage module)
+  command:
+    cmd: emerge gentoolkit
+    creates: "{{ prefix_location }}/usr/bin/equery"
+
+- name: Create repos directory
+  file:
+    path: "{{ prefix_location }}/etc/portage/repos.conf"
+    state: directory
+    mode: '0755'
+
+- name: Install eselect-repository
+  portage:
+    package: eselect-repository
+    state: present
+
+# We need git in order to add Gentoo overlays hosted on git repositories.
+- name: Install git
+  portage:
+    package: dev-vcs/git
+    state: present
+
+- name: Add our overlay configuration
```

Comment:
There is some protection in the task itself:
```
- name: Add our overlay configuration
  command:
    cmd: "eselect repository add {{ custom_overlay_name }} {{ custom_overlay_source }} {{ custom_overlay_url }}"
    creates: "{{ gentoo_prefix_path }}/etc/portage/repos.conf/eselect-repo.conf"
```
The last line will ensure that this only gets run when there is no eselect-repo.conf yet. This file should have been created if you have run this playbook before. However, if you for instance run this on an Prefix installation where you already added another repo, this task won't run either. But to completely solve this, we would need to check the contents of the file and see if our repo configuration is already in there, which is a bit more tricky. It will have to check if the following section is included in the file, and all settings correspond to the parameters of the playbook:
```
[eessi]
location = /cvmfs/pilot.eessi-hpc.org/test/gentoo/2020/var/db/repos/eessi
sync-type = git
sync-uri = https://github.com/EESSI/gentoo-overlay.git

```

Reply:
By the way, I also tested this by running the playbook multiple times, and for instance only adding a new package to our package set to verify if this gets installed. That worked fine for me. Did you get this error when manually trying to repeat this eselect command, or did it happen while running the playbook?

Reply:
I got the error by running manually. 
I haven't noticed the `creates:` section in the task. That's my bad.
This somehow protects the task, but as you said it is not optimal.
What about considering to have a protector task that checks whether the eessi repo is installed.
Something in the lines of

```
- name: Check if eessi repo is installed
   command: eselect repository list -i eessi
   register: eessi_repository_installed

- name: Add eessi overlay configuration
  command:
    cmd: "eselect repository add {{ custom_overlay_name }} {{ custom_overlay_source }} {{ custom_overlay_url }}"
    when: not eessi_repository_installed.stdout.find('eessi')
```

Reply:
I think this is indeed a better approach for this check. I'll modify this, thanks!